supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiAsyncBackendJobExecutionActor.scala

```diff
@@ -827,16 +832,28 @@ class PipelinesApiAsyncBackendJobExecutionActor(override val standardParams: Sta
         throw new RuntimeException(s"handleExecutionSuccess not called with RunStatus.Success. Instead got $unknown")
     }
 
-  override def getStartAndEndTimes(runStatus: StandardAsyncRunState): Option[StartAndEndTimes] =
-    runStatus match {
-      case terminalRunStatus: TerminalRunStatus if terminalRunStatus.eventList.nonEmpty =>
-        val offsetDateTimes = terminalRunStatus.eventList.map(_.offsetDateTime)
-        val cpuStart = terminalRunStatus.eventList.find(event =>
-          event.name.matches("""^Worker[\s\"\\]+google-pipelines-worker-[a-zA-Z0-9\s\"\\]+assigned in.*""")
+  override def getStartAndEndTimes(runStatus: StandardAsyncRunState): Option[StartAndEndTimes] = {
+    // Intuition:
+    // "job start" is the earliest event time across all events.
+    // "cpuStart" is obtained from the cost helper. It should be the event time where the user VM started spending money.
+    // "jobEnd" is obtained from the cost helper, falling back to the last event time, falling back to now.
+    // We allow fallbacks for end times to account for in progress runs, but generally the end time is only known once we've reached a terminal status.
+    val jobStart: Option[OffsetDateTime] =
+      runStatus.eventList.headOption.map(_ => runStatus.eventList.map(_.offsetDateTime).min)
+    val maxEventTime: Option[OffsetDateTime] =
+      runStatus.eventList.headOption.map(_ => runStatus.eventList.map(_.offsetDateTime).max)
+    costHelper match {
+      case Some(helper) =>
+        val cpuStart: Option[OffsetDateTime] = helper.vmStartTime
+        val jobEnd: OffsetDateTime = helper.vmEndTime.orElse(maxEventTime).getOrElse(OffsetDateTime.now())
```

Comment:
See my other comment re: bard. It's a slight change to this function, but I think at the end of the day bard will see the same times that it is currently. 

I am open to changing this fallback though. `now()` makes sense to me because of how we're using the end time & when we're emitting it, but I could also see `None` and/or the max event time being preferable.

Reply:
I'm waffling, for cost capping having some failsafe to ensure we have ANY value feels important - but how likely is it really that we don't even have `maxEventTime`? If we don't have that, we have nothing. I'm fine either way.